api/src/main/java/com/epam/esm/dao/impl/TagDaoImpl.java

```diff
@@ -0,0 +1,78 @@
+package com.epam.esm.dao.impl;
+
+import com.epam.esm.dao.TagDao;
+import com.epam.esm.dao.mapper.TagRowMapper;
+import com.epam.esm.domain.Tag;
+import com.epam.esm.exception.ResourceNotFoundException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.jdbc.support.KeyHolder;
+import org.springframework.stereotype.Repository;
+
+import java.sql.PreparedStatement;
+import java.util.List;
+import java.util.Objects;
+
+@Repository
+@RequiredArgsConstructor
+public class TagDaoImpl implements TagDao {
+
+    private static final String GET_BY_ID = "SELECT tag.id t_id, tag.name t_name FROM `tag` WHERE tag.id=?";
+    private static final String GET_ALL = "SELECT tag.id t_id, tag.name t_name FROM `tag`";
+    private static final String INSERT_TAG = "INSERT INTO `tag` (`name`) VALUES (?)";
+    private static final String DELETE_TAG_LINK = "DELETE FROM `certificate_has_tag` WHERE tag_id=?";
+    private static final String DELETE_TAG = "DELETE FROM `tag` WHERE id=?";
+    private static final String GET_BY_NAME = "SELECT tag.id t_id, tag.name t_name FROM `tag` WHERE tag.name=?";
+
+    private final JdbcTemplate jdbcTemplate;
+    private final TagRowMapper tagRowMapper;
+
+    @Override
+    public Tag getOne(Long id) {
+        try {
+            return jdbcTemplate.queryForObject(GET_BY_ID, new Object[]{id}, tagRowMapper);
+        } catch (EmptyResultDataAccessException e) {
+            throw new ResourceNotFoundException("Resource not found", e, id);
```

Comment:
NIT: Resource is REST concept that don't belong to data access layer. I'd personally remove this exceptions from here and handle'em on service or even controller layer.